AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -158,11 +163,55 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
   (void)completionHandler;
 }
 
-#if TARGET_OS_OSX
-- (void)dealloc {
-}
+#pragma mark - MSDataStore Implementation
+- (void)createWithPartition:(NSString *)partition
+                 documentId:(NSString *)documentId
+                   document:(id<MSSerializableDocument>)document
+               writeOptions:(MSWriteOptions *)__unused writeOptions
+          completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
 
-#endif
+  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
+                   partitions:@[ partition ]
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+
+              // If error getting token.
+              if (error && !tokenResponses) {
+                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token: %@", [error description]);
+                completionHandler([[MSDocumentWrapper alloc] initWithError:error documetnId:documentId]);
+              }
+
+              // Create http client.
+              MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+
+              NSError *serializationError;
+              NSData *body = [NSJSONSerialization dataWithJSONObject:[document serializeToDictionary] options:0 error:&serializationError];
+              if (!body && serializationError) {
+                MSLogError([MSDataStore logTag], @"Error serializing data: %@", [serializationError description]);
+                completionHandler([[MSDocumentWrapper alloc] initWithError:serializationError documetnId:documentId]);
+              }
+
+              // Call CosmosDb.
+              [MSCosmosDb performCosmosDbAsyncOperationWithHttpClient:(MSCosmosDbIngestion *)cosmosDbIngestion
+                                                          tokenResult:tokenResponses.tokens[0]
+                                                           documentId:@""
+                                                             httpVerb:@"POST"
+                                                                 body:body
+                                                    completionHandler:^(NSData *_Nonnull data, NSError *_Nonnull cosmosDbError) {
+                                                      if (cosmosDbError || data != nil) {
+
+                                                        // Create a document.
+                                                        MSDocumentWrapper *docWrapper =
+                                                            [[MSDocumentWrapper alloc] initWithDeserializedValue:document
+                                                                                                       partition:partition
+                                                                                                      documetnId:documentId
+                                                                                                            etag:@"myEtag"
```

Comment:
Why is the `etag` a constant? Any TODO for that one?
Also, please use `eTag` instead of `etag` since we are using this in other places.